src/commands/deployAzurePipeline/configureCicdPipeline/models/aksClusterNode.ts

```diff
@@ -0,0 +1,13 @@
+import { CloudExplorerV1 } from "vscode-kubernetes-tools-api";
+
+export class aksClusterNode{

```

Comment:
💡 I think if you can declare an interface that is a nice clean way to contract bound `Object` with well defined expected types. Explore this: https://www.typescriptlang.org/docs/handbook/interfaces.html

Reply:
+1. It needs to be an interface.

Reply:
Actually, would it make more sense to use `GenericResource` interface from `azure-arm-resource` package/ This is a very generic definition. AKS extension already takes dependency on this package so we won't adding a new dependency.

We might have to extend it though for adding `subscriptionid`

Reply:
💡 I am unsure of using `GenericResource` in this context, and how you guys think it caters this comment here: https://github.com/Azure/vscode-aks-tools/pull/42#discussion_r499921565 

Possibly missing a context so please feel free to add more details? : )

Reply:
@Tatsinnit the goal is to define a contract which is de-coupled from AKS extension's tree node structure. We can either define a custom resource model which can contain properties which are currently required by DTA extension.
Another solution would be to use this `GenericResource` model. The benefits are:
1. It can be used for any other resource as well. For example, if DTA has to integrate with Web App resource, we will not define to update the contract. This will save us from unnecessarily updating AKS extension when we integrate with App service extension.
2. If we need to pass more properties of the cluster to DTA extension in future, we can do so without changing the contract. GenericResource is more flexible and has other fields which can be populated if need be.

Reply:
@Tatsinnit Does this answer your question?

Reply:
@bishal-pdMSFT Now after [this commit](https://github.com/Azure/vscode-aks-tools/pull/42/commits/c521426abb70e8043fee5be826d8c12b01daa8f7), it makes sense, and more aligned to what I was thinking. I will let Ivan kindly do a quick take.

https://github.com/Azure/vscode-aks-tools/pull/42/commits/c521426abb70e8043fee5be826d8c12b01daa8f7